design/pool-configuration.md

```diff
@@ -0,0 +1,552 @@
+# MetalLB configuration options
+
+## Summary
+
+Following @johananl's initiative in [this issue](https://github.com/metallb/metallb/issues/630)
+this enhancement tries to collect and classify various configuration enhancement requests
+requested over time.
+
+## Motivation
+
+The way advanced users consume MetalLB is evolving and exposed a need for more complex configuration.
+For example, the need for isolation and multi-tenancy induced requests for being able to advertise some specific address pools from a subset of the nodes, or
+to allocate them only to a specific namespace, to mention some.
+
+As part of the discussion over an initial version of this proposal we also decided to rethink the MetalLB datamodel, in order to separate the different concerns
+handled by MetalLB. This separation is intended to be implemented dropping the configmap support and using Kubernetes CRDs.
+
+### Goals
+
+* Collect and classify all the requests coming from different issues, in order to have a better understanding
+  and an organic implementation, avoiding focus on and implementation of only a specific feature, risking causing conflicts afterwards.
+* Provide a list of the possible configuration enhancements and highlight dependencies and conflicts among them (if any).
+* Highlight the cases where the solution must be BGP or L2 specific
+* Provide a new version of the datamodel, splitting the configuration in different resources responsible to address different concerns, and
+use that as a base for the new configuration degrees proposed in this document.
+
+### Non-Goals
+
+* Covering only a specific use case.
+* Providing a L2 only or BGP only solution where the problem might be expressed in a more generic way.
+
+## New data model
+
+Before addressing the new concerns, we will propose a new scheme for configuring MetalLB, which will be used as a starting point
+in the following parts of the proposal. For this reason, it makes sense to have it described in this proposal instead of a separate one.
+
+The new data model will be expressed in terms of kubernetes CRDs, deprecating the configuration via the metallb's ConfigMap.
+
+In the new data model, the following entities will be defined (more details on the implementation section):
+
+* `AddressPool`, responsible of the ip allocation to services
+* `L2Advertisement`, describing the intention to advertise an IP coming from an `AddressPool` via L2
```

Comment:
>the intention to advertise an IP

An IP or a **prefix**? If we're really separating IP allocation from advertisement, I assume I should be able to aggregate multiple IPs in a `BGPAdvertisement` as a `/28` for example.